lib/ts/fetch.ts

```diff
@@ -279,15 +279,20 @@ export default class AuthHttpRequest {
                     logDebugMessage("doRequest: rid header was already there in request");
                 }
 
+                await setTokenHeaders(clonedHeaders);
```

Comment:
EDIT: We actually do want to do the above cause if we don't then migration would be a problem (from cookie to header based). So what we will do is to add these credentials and anti-csrf header all the time, but then after a month or so of release, we will set the autoAddCredentials to false by default in case mode is headers